astroquery/heasarc/core.py

```diff
@@ -252,7 +252,7 @@ def _parse_result(self, response, verbose=False):
 
         try:
             data = BytesIO(response.content)
-            return Table_read(data, hdu=1)
+            return Table_read(data)
```

Comment:
you'll need to get rid of the `Table_read` workaround, too.

Reply:
Ok, yes. 
Though I think it will be still needed for the fallback, old interface has no VOTable.

I copied the code from #2333 to a test:

https://github.com/oda-hub/astroquery/blob/b09c8b588bafa0078c2a6996b251af669a1a894d/astroquery/heasarc/tests/test_heasarc_remote.py#L108-L123

Currently it fails while trying to `_convert_record2fits`. Maybe if it starts with fits it's more compatible. Some investigation is needed.

Reply:
> you'll need to get rid of the `Table_read` workaround, too.

I have been trying to work on this conversion to VOTable on my own. Is there any other reason why we need to keep the `Table_read` function in other than silence the unit parsing?

For refernce:

```
def Table_read(*args, **kwargs):
    if commons.ASTROPY_LT_5_1:
        return Table.read(*args, **kwargs)
    else:
        return Table.read(*args, **kwargs, unit_parse_strict='silent')
```

I only ask since I in some testing with some of my own changes I've cut out this function and most tables work with VOTable perfectly fine without warnings. 

I can submit a separate PR with more info if that would be more appropriate.